src/encoded/static/components/dataset.js

```diff
@@ -2506,6 +2620,43 @@ ReplicationTimingSeries.contextTypes = {
 globals.contentViews.register(ReplicationTimingSeries, 'ReplicationTimingSeries');
 
 
+/**
+ * Wrapper component for differentiation series pages.
+ */
+const DifferentiationSeries = ({ context }, reactContext) => {
+    const seriesType = context['@type'][0];
+    const seriesTitle = reactContext.profilesTitles[seriesType] || '';
+    const findDifferentiations = context.related_datasets.map((dataset) => (computeDifferentiation(dataset) ? 1 : 0)).reduce((a, b) => a + b, 0);
+    const differentiationTableColumns = findDifferentiations > 0 ? differentiationTableColumnsWithTime : differentiationTableColumnsWithoutTime;
```

Comment:
I was thinking another way to handle this is to use the `hide` property for SortTable. Whether that’s easier or harder than what you have varies by case, but it does reduce the code duplication for the other columns.

Reply:
I like this idea. I did think it was quite a bit of extra code to define two sets of columns for when one variable existed vs not.